intermediate_source/FSDP_tutorial.rst

```diff
@@ -46,6 +46,15 @@ At a high level FSDP works as follow:
 * Run reduce_scatter to sync gradients
 * Discard parameters. 
 
+The key insight behind full parameter sharding is that we can decompose the all-reduce operations in DDP into separate reduce-scatter and all-gather operations.
```

Comment:
I am not sure that this is the correct statement.

Even though an all-reduce can be decomposed as a reduce-scatter and all-gather, the current phrasing might suggest that DDP's _gradient_ all-reduce is being decomposed into a _gradient_ reduce-scatter and _gradient_ all-gather. However, FSDP actually all-gathers parameters.

Whether or not this decomposition of all-reduce into reduce-scatter and all-gather is the _key_ insight is not obvious to me. If we show this decomposition, we probably want more exposition.

Reply:
I agree. Adding this picture would demand a clear explanation. 

I am unsure what to write. If you can suggest something or direct me to where I can read about this topic, that'd be very helpful.

Reply:
@awgu can you help @ChanBong with this?

Reply:
Maybe something like the following:

One way to view FSDP's sharding is to decompose the DDP gradient all-reduce into reduce-scatter and all-gather. In particular, FSDP reduce-scatters gradients such that each rank has a shard of the gradients in backward, updates the corresponding shard of the parameters in the optimizer step, and all-gathers them in the next forward.

< Figure >

Reply:
Sounds good. Thanks